src/movement/control_system.cpp

```diff
@@ -187,265 +195,355 @@ void ControlSystem::InputControlMessage(robosub::control msg)
+    current_state.yaw_left_derivative = current_derivative[5] / _PI_OVER_180;
+
+    current_state.forward_rotation_control = rotation_control[0];
+    current_state.strafe_left_rotation_control = rotation_control[1];
+    current_state.dive_rotation_control = rotation_control[2];
+    current_state.roll_right_rotation_control = rotation_control[3] / _PI_OVER_180;
+    current_state.pitch_down_rotation_control = rotation_control[4] / _PI_OVER_180;
+    current_state.yaw_left_rotation_control = rotation_control[5] / _PI_OVER_180;
+
+    current_state.forward_translation_control = translation_control[0];
+    current_state.strafe_left_translation_control = translation_control[1];
+    current_state.dive_translation_control = translation_control[2];
+    current_state.roll_right_translation_control = translation_control[3] / _PI_OVER_180;
+    current_state.pitch_down_translation_control = translation_control[4] / _PI_OVER_180;
+    current_state.yaw_left_translation_control = translation_control[5] / _PI_OVER_180;
 
     return current_state;
 }
-void ControlSystem::PublishThrusterMessage()
-{
-    pub->publish(tp);
-}
 
-/*
-string ControlSystem::statestring(control_packet::State state)
+std::string ControlSystem::state_to_string(uint8_t state)
 {
-    string s;
     switch (state)
     {
-        case robosub::control::STATE_NONE:     s = "NONE"; break;
-        case robosub::control::STATE_ABSOLUTE: s = "ABSOLUTE"; break;
-        case robosub::control::STATE_RELATIVE: s = "RELATIVE"; break;
-        case robosub::control::STATE_ERROR:    s = "ERROR"; break;
-        default: s = "UNKNOWN"; break;
+        case robosub::control::STATE_NONE:
+            return "No State";
+            break;
+        case robosub::control::STATE_ERROR:
+            return "Error State";
+            break;
+        case robosub::control::STATE_ABSOLUTE:
+            return "Absolute State";
+            break;
+        case robosub::control::STATE_RELATIVE:
+            return "Relative State";
+            break;
+        default:
+            return "Unknown State";
+            break;
     }
+    return "Unknown State";
+}
 
-    return s;
+/**
+ * Publishes a thruster command.
+ *
+ * @return None.
+ */
+void ControlSystem::PublishThrusterMessage()
+{
+    pub->publish(tp);
 }
-*/
 
-VectorXd ControlSystem::motor_control(Vector12d state)
+/**
+ * Calculates the motor control signal.
+ *
+ * @return The control signal C to send to the thrusters.
+ */
+VectorXd ControlSystem::calculate_motor_control()
 {
-    //INFO("running motor control function");
-    //calcluate errors
-    Vector3d err_translation = goals.segment<3>(0) - state.segment<3>(0);
+    /*
+     * Calculate the error between the current state and the goal. If the state
+     * is set to error, override the error calculation to assume that our
+     * current state is correct and no modification is required.
+     */
+    ROS_INFO_STREAM("Current State: \n" << state_vector);
```

Comment:
This is being done every time the node spins. This will also clutter the terminal. I suggest using `DEBUG` rather than `INFO`